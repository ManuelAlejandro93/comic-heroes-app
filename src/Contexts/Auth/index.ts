export * from './AuthContext';
export * from './Reducer';
export * from './Hooks';
export * from './Interfaces';
