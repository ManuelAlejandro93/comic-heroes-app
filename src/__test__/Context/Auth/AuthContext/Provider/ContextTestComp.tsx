import { useContext } from 'react';
import { AuthContext } from '@/Contexts';

export const ContextTestComp = () => {
  const { authState, onLogin, onLogout } = useContext(AuthContext);
  return (
    <div>
      <h1>Pruebas en Auth Context Provider</h1>
      <p>{`${authState.name}`}</p>
      <p>{`${authState.logged}`}</p>
      <button onClick={() => onLogin()}>click on login</button>
      <button onClick={() => onLogout()}>click on logout</button>
    </div>
  );
};
