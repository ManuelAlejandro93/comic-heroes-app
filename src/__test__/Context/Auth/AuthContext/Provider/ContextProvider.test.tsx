import { AuthProvider } from '@/Contexts';
import { act, fireEvent, render, screen } from '@testing-library/react';
import { ContextTestComp } from './ContextTestComp';

describe('Pruebas en el contexto AuthProvider', () => {
  test('debe proveer coincidir con el snapshot', () => {
    const { container } = render(
      <AuthProvider>
        <ContextTestComp></ContextTestComp>
      </AuthProvider>
    );
    expect(container).toMatchSnapshot();
  });

  test('debe traer con useContext el valor del estado auth inicial', () => {
    const { container } = render(
      <AuthProvider>
        <ContextTestComp></ContextTestComp>
      </AuthProvider>
    );
    expect(container.querySelector('p')?.innerHTML).toBe('null');
    expect(screen.getByText('false').innerHTML).toBe('false');
  });

  test('Al darle click en el botÃ³n "click en login" deben cambiar los valores a valores de login.', () => {
    const { container } = render(
      <AuthProvider>
        <ContextTestComp></ContextTestComp>
      </AuthProvider>
    );

    expect(container.querySelector('p')?.innerHTML).toBe('null');

    const botonLogin = container.querySelectorAll('button')[0];
    // const botonLogout = container.querySelectorAll('button')[1];

    act(() => {
      fireEvent.click(botonLogin);
    });

    expect(container.querySelectorAll('p')[0]?.innerHTML).toBe('Manuel');
    expect(container.querySelectorAll('p')[1]?.innerHTML).toBe('true');
  });

  test('Al loggerase y luego desloggearse deben cambiar los valores de valores de loggeado a valores de desloggeado.', () => {
    const { container } = render(
      <AuthProvider>
        <ContextTestComp></ContextTestComp>
      </AuthProvider>
    );

    const botonLogout = container.querySelectorAll('button')[1];

    expect(container.querySelectorAll('p')[0]?.innerHTML).toBe('Manuel');
    expect(container.querySelectorAll('p')[1]?.innerHTML).toBe('true');

    act(() => {
      fireEvent.click(botonLogout);
    });

    expect(container.querySelectorAll('p')[0]?.innerHTML).toBe('');
    expect(container.querySelectorAll('p')[1]?.innerHTML).toBe('false');
  });
});
