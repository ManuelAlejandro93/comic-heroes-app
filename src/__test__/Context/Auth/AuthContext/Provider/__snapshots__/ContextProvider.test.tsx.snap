// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el contexto AuthProvider debe proveer coincidir con el snapshot 1`] = `
<div>
  <div>
    <h1>
      Pruebas en Auth Context Provider
    </h1>
    <p>
      null
    </p>
    <p>
      false
    </p>
    <button>
      click on login
    </button>
    <button>
      click on logout
    </button>
  </div>
</div>
`;
